### PR TITLE
feat: add support for text generation tasks

### PR DESCRIPTION
**Pull request by heiruwu**
Because

- support text generation tasks in VDP

This commit

- add support for text generation tasks

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ type MaxBatchSizeConfig struct {
 	Ocr                  int `koanf:"ocr"`
 	InstanceSegmentation int `koanf:"instancesegmentation"`
 	SemanticSegmentation int `koanf:"semanticsegmentation"`
+	TextGeneration		 int `koanf:"textgeneration"`
```
Comment (Phelan164) [inline config/config.go:98]: please format the code

Comment (heiruwu) [reply config/config.go:98]: will do

File: pkg/handler/payload.go [hunk 2]
```diff
@@ -267,3 +282,48 @@ func parseImageFormDataTextToImageInputs(req *http.Request) (textToImageInput []
 		Samples:  1,
 	}}, nil
 }
+
+func parseTextFormDataTextGenerationInputs(req *http.Request) (textGeneration []triton.TextGenerationInput, err error) {
+	prompts := req.MultipartForm.Value["prompt"]
+	if len(prompts) != 1 {
+		return nil, fmt.Errorf("invalid input")
```
Comment (Phelan164) [inline pkg/handler/payload.go:289]: nit: maybe a more specific message such as "the model do not support batch, only single prompt is allowed"

File: pkg/handler/payload.go [hunk 3]
```diff
@@ -267,3 +282,48 @@ func parseImageFormDataTextToImageInputs(req *http.Request) (textToImageInput []
...
+	}
+
+	output_len := req.MultipartForm.Value["output_len"]
+	top_k := req.MultipartForm.Value["top_k"]
+	seed := req.MultipartForm.Value["seed"]
+
+	_output_len := 100
```
Comment (Phelan164) [inline pkg/handler/payload.go:296]: Golang we do not use the first underscore, we use the camelCase convention

File: pkg/handler/payload.go [hunk 4]
```diff
@@ -267,3 +282,48 @@ func parseImageFormDataTextToImageInputs(req *http.Request) (textToImageInput []
...
 	}}, nil
 }
+
+func parseTextFormDataTextGenerationInputs(req *http.Request) (textGeneration []triton.TextGenerationInput, err error) {
+	prompts := req.MultipartForm.Value["prompt"]
+	if len(prompts) != 1 {
+		return nil, fmt.Errorf("invalid input")
+	}
+
+	output_len := req.MultipartForm.Value["output_len"]
```
Comment (Phelan164) [inline pkg/handler/payload.go:292]: should use cammelCase

Comment (heiruwu) [reply pkg/handler/payload.go]: done

File: pkg/handler/payload.go (hunk 3 above)
Comment (heiruwu) [reply pkg/handler/payload.go]: done

File: pkg/handler/payload.go (hunk 2 above)
Comment (heiruwu) [reply pkg/handler/payload.go]: nice catch, revised

File: pkg/handler/payload.go [hunk 5]
```diff
@@ -172,6 +172,21 @@ func parseTexToImageRequestInputs(req *modelPB.TriggerModelInstanceRequest) (tex
 	return textToImageInputs, nil
 }
 
+func parseTexGenerationRequestInputs(req *modelPB.TriggerModelInstanceRequest) (textGenerationInput []triton.TextGenerationInput, err error) {
+	var textGenerationInputs []triton.TextGenerationInput
+	for _, taskInput := range req.TaskInputs {
+		textGenerationInputs = append(textGenerationInputs, triton.TextGenerationInput{
+			Prompt:        taskInput.GetTextGeneration().Prompt,
+			OutputLen:     *taskInput.GetTextGeneration().OutputLen,
```
[the remark this answers is not in the document]
Comment (heiruwu) [reply pkg/handler/payload.go]: done

**Review by Phelan164 — APPROVED** [review]
LGTM

**Pull request merged**